traversal/planner/GraphPlanner.java

```diff
@@ -453,9 +421,12 @@ public void execute() {
             queue.add(start);
             while (!queue.isEmpty()) {
                 PlannerVertex<?> vertex = queue.iterator().next();
-                List<PlannerEdge.Directional<?, ?>> outgoing = vertex.outs().stream()
-                        .filter(e -> !e.hasInitialValue() && !(e.isSelfClosure() && e.direction().isBackward()))
-                        .sorted(comparing(e -> e.costLastRecorded)).collect(toList());
+                List<PlannerEdge.Directional<?, ?>> outgoing = new ArrayList<>();
+                vertex.outs().stream().filter(e -> !e.hasInitialValue()).sorted(comparing(e -> e.costLastRecorded))
+                        .forEachOrdered(edge -> {
+                            if (!(edge.isSelfClosure() && edge.direction().isBackward())) outgoing.add(edge);
+                            else edge.setInitialUnselected();
+                        });
```

Comment:
each `Variable` must have a valid initial value. In the old code we actually relied on relatively implicit initialisation to 0 in order to give each variable a hint. We now explicitly set initial values in all cases - the new case is `edge.setInitialUnselected()`